### ISSUE TITLE
Error: Uncaught (in promise): InvalidStateError: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.Error: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing. 

### ISSUE THREAD

**Issue by mohsenone** [johannesjo/super-productivity#886]
Error: Uncaught (in promise): InvalidStateError: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.Error: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.    at Proxy.n (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2346445)    at Proxy.s (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2347527)    at e. (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2355867)    at Generator.next ()    at a (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2355205)    at l.invoke (file:///snap/superproductivity/1004/resources/app.asar/dist/polyfills.bbc1c0564b6a5d42ba73.js:1:6572)    at Object.onInvoke (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:1555412)    at l.invoke (file:///snap/superproductivity/1004/resources/app.asar/dist/polyfills.bbc1c0564b6a5d42ba73.js:1:6512)    at i.run (file:///snap/superproductivity/1004/resources/app.asar/dist/polyfills.bbc1c0564b6a5d42ba73.js:1:1984)    at file:///snap/superproductivity/1004/resources/app.asar/dist/polyfills.bbc1c0564b6a5d42ba73.js:1:16758

### Steps to Reproduce

1.
2.
3.
4.

### Stacktrace
```
function (webpack:///node_modules/idb/build/esm/wrap-idb-value.js:150:25)
function (webpack:///node_modules/idb/build/esm/index.js:65:24)
constructor (webpack:///src/app/core/persistence/database.service.ts:49:49)
Ur (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2355205)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:368:25)
onInvoke (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28281:32)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:367:51)
run (webpack:///node_modules/zone.js/dist/zone-evergreen.js:130:42)
zone.scheduleMicroTask(source, (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1272:35)
scheduleResolveOrReject (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1209:30)
makeResolver (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1116:16)
Ur (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:1400545)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:368:25)
onInvoke (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28281:32)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:367:51)
run (webpack:///node_modules/zone.js/dist/zone-evergreen.js:130:42)
zone.scheduleMicroTask(source, (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1272:35)
invokeTask (webpack:///node_modules/zone.js/dist/zone-evergreen.js:402:30)
onInvokeTask (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28269:32)
```

### Meta Info
META: SP6.3.3 Electron – en-US – Linux x86_64 – Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/6.3.3 Chrome/85.0.4183.121 Electron/10.2.0 Safari/537.36

### Actions Before Error
```
1612691651492: [Project] Update Work End 
1612691651493: [Tag] Update Work End for Tag 
1612691651503: [Persistence] Save to DB 
1612691651504: [Persistence] Save to DB 
1612691651504: [Persistence] Save to DB 
1612691652489: [Task] Add time spent 
1612691652491: [Project] Update Work End 
1612691652493: [Tag] Update Work End for Tag 
1612691652501: [Persistence] Save to DB 
1612691652502: [Persistence] Save to DB 
1612691652502: [Persistence] Save to DB 
1612691653096: [Pomodoro] Pause Pomodoro 
1612691653098: [Task] UnsetCurrentTask 
1612691653101: [Pomodoro] Pause Pomodoro 
1612775774617: [Tag] Update Break Time for Tag 
1612775774667: [Persistence] Save to DB 
1612780288863: [Layout] Show AddTaskBar 
1612780292217: [Task][Issue] Add Task 
1612780292231: [Persistence] Save to DB 
1612780292231: [Persistence] Save to DB 
1612780293811: [Layout] Hide AddTaskBar 
1612781733814: [Tag] Update Break Time for Tag 
1612781733822: [Persistence] Save to DB 
1612800763647: [Task] Update Task 
1612800763991: [Persistence] Save to DB 
1612800766574: [Task] SetSelectedTask 
1612800785284: [Task] Update Task 
1612800785285: [Persistence] Save to DB 
1612800785294: [Task] Update Task 
1612800785295: [Persistence] Save to DB
```

**Comment by johannesjo** [comment]
Thanks for reporting! Could you maybe provide a little bit more context?

**Comment by AlaricLightin** [comment]
I have received the same bug today. For me it seems as I don't do anything significant. I have switched window to Superproductivity and application have crashed.

### Stacktrace
```
function (webpack:///node_modules/idb/build/esm/wrap-idb-value.js:150:25)
function (webpack:///node_modules/idb/build/esm/index.js:65:24)
constructor (webpack:///src/app/core/persistence/database.service.ts:49:49)
Ur (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:2355205)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:368:25)
onInvoke (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28281:32)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:367:51)
run (webpack:///node_modules/zone.js/dist/zone-evergreen.js:130:42)
zone.scheduleMicroTask(source, (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1272:35)
scheduleResolveOrReject (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1209:30)
makeResolver (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1116:16)
Ur (file:///snap/superproductivity/1004/resources/app.asar/dist/main.839fd4a6fbc40d5fd73d.js:1:1400545)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:368:25)
onInvoke (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28281:32)
invoke (webpack:///node_modules/zone.js/dist/zone-evergreen.js:367:51)
run (webpack:///node_modules/zone.js/dist/zone-evergreen.js:130:42)
zone.scheduleMicroTask(source, (webpack:///node_modules/zone.js/dist/zone-evergreen.js:1272:35)
invokeTask (webpack:///node_modules/zone.js/dist/zone-evergreen.js:402:30)
onInvokeTask (webpack:///node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:28269:32)
```

### Meta Info
META: SP6.3.3 Electron – en-US – Linux x86_64 – Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/6.3.3 Chrome/85.0.4183.121 Electron/10.2.0 Safari/537.36

### Actions Before Error
```
1613036112594: [Task] Add time spent 
1613036112598: [Project] Update Work End 
1613036112599: [Tag] Update Work End for Tag 
1613036112610: [Persistence] Save to DB 
1613036112610: [Persistence] Save to DB 
1613036112610: [Persistence] Save to DB 
1613036113594: [Task] Add time spent 
1613036113598: [Project] Update Work End 
1613036113601: [Tag] Update Work End for Tag 
1613036113618: [Persistence] Save to DB 
1613036113619: [Persistence] Save to DB 
1613036113620: [Persistence] Save to DB 
1613036114594: [Task] Add time spent 
1613036114599: [Project] Update Work End 
1613036114601: [Tag] Update Work End for Tag 
1613036114614: [Persistence] Save to DB 
1613036114614: [Persistence] Save to DB 
1613036114615: [Persistence] Save to DB 
1613036115594: [Task] Add time spent 
1613036115598: [Project] Update Work End 
1613036115599: [Tag] Update Work End for Tag 
1613036115610: [Persistence] Save to DB 
1613036115612: [Persistence] Save to DB 
1613036115613: [Persistence] Save to DB 
1613036122723: [Task] Add time spent 
1613036122726: [Project] Update Work End 
1613036122727: [Tag] Update Work End for Tag 
1613036122731: [Persistence] Save to DB 
1613036122732: [Persistence] Save to DB 
1613036122732: [Persistence] Save to DB
```

**Comment by johannesjo** [comment]
Thanks! Looks like a duplicate of #884

**Comment by iamtburger** [comment]
> Thanks for reporting! Could you maybe provide a little bit more context?

Well, I can't really add anything to that. I basically just switched from one window to another and then this popped up.
I did nothing special, Unfortunately I don't remember if it was actively tracking one of my tasks and I switched to pause, or it was paused and I wanted to start tracking.
If there is something specific that could help, let me know!
